---
title: "QTL analysis of behavioral traits in Diversity Outbred Mice"
source: Rmd
---

```{r message=FALSE, echo=FALSE}
library(qtl2)
library(qtl2convert)
library(qtl2db)
#library(qtl2plot)
```

This tutorial will take you through the process of mapping a QTL and searching for candidate genes.

The data comes from a behavioral study in which Diversity Outbred (DO) mice were we characterized *283* male and female DO mice using open-field, lightâ€“dark box, tail-suspension and visual-cliff avoidance tests to generate 38 behavioral measures. We identified several quantitative trait loci (QTL) for these traits with support intervals ranging from 1 to 3 Mb in size. These intervals contain relatively few genes (ranging from 5 to 96). For a majority of QTL, using the founder allelic effects together with whole genome sequence data, we could further narrow the positional candidates. Several QTL replicate previously published loci. Novel loci were also identified for anxiety- and activity-related traits. Half of the QTLs are associated with wild-derived alleles, confirming the value to behavioral genetics of added genetic diversity in the DO. In the presence of wild-alleles we sometimes observe behaviors that are qualitatively different from the expected response. Our results demonstrate that high-precision mapping of behavioral traits can be achieved with moderate numbers of DO animals, representing a significant advance in our ability to leverage the mouse as a tool for behavioral genetics


There are two publications associated with these data. These are [Recla JM, et al. (2014) (<i>Mamm Genome</i> 25:211-222)](https://www.ncbi.nlm.nih.gov/pubmed/24700285) and 
[Logan RW, et al. (2014) (<i>Genes Brain Behav</i> 12:424-437)](https://www.ncbi.nlm.nih.gov/pubmed/23433259)


## QTL mapping workflow
![](../figs/mapping-workflow.png)

### Trait heritability
Heritability estimates for the 38 behavioral measurements were calculated from the progenitor strain data using variance components from a mixed model with strain as a random effect. 
![](../figs/Table1_h2.png)
Heritabilities can also be computed on measurements collected using the DO by leveraging the information containted in the kinship matrix.



### Load and explore the data

```{r load_data}
do.cross <- read_cross2("../data/DO_2014/DO2014.json")
summary(do.cross)
```


#### The Marker Map  
The markers are from a genotyping array called the Mouse Universal Genotyping Array (MUGA) and contains 7,856 SNP markers. Marker locations for the MUGA and other mouse arrays are available from [The Jackson Laboratory's FTP site](ftp://ftp.jax.org/MUGA).

#### Phenotype distributions

**Open field (OF)**

```{r plot_phenotypes_OF, warning=FALSE, message=FALSE, fig.height=8, fig.width=8, echo=FALSE}
library(ggplot2)
library(reshape)
library(GGally)
pheno <- as.data.frame(do.cross$pheno)
OF <- grep(colnames(pheno), pattern = "^OF")
ggpairs(pheno[,OF])
```


**Light Dark (LD)**

```{r plot_phenotypes_LD, warning=FALSE, message=FALSE, fig.height=8, fig.width=8, echo=FALSE}
library(ggplot2)
library(reshape)
library(GGally)
pheno <- as.data.frame(do.cross$pheno)
LD <- grep(colnames(pheno), pattern = "^LD")
ggpairs(pheno[,LD])
```

**Tail Suspension Test (TST)**

```{r plot_phenotypes_TST, warning=FALSE, message=FALSE, fig.height=8, fig.width=8, echo=FALSE}
library(ggplot2)
library(reshape)
library(GGally)
pheno <- as.data.frame(do.cross$pheno)
TS <- grep(colnames(pheno), pattern = "^TS")
ggpairs(pheno[,TS])
```
